### DISCUSSION TITLE
[platform/azure] - Allow renovate bot to bypass Azure PR branch policy

### DISCUSSION THREAD

**Discussion by marcmognol** [renovatebot/renovate#28502] [Suggest an Idea]
### Tell us more.

Dear renovate team,

In our Azure repositories we set some branch policies for our PR to add at least 2 reviewers approvals.
We can give Renovate Bot user the permission to bypass policies:

![image](https://github.com/renovatebot/renovate/assets/8171300/82d7f5af-b778-487e-ab01-30263ad04355)

But, to complete the PR, it is required to specifiy the following attributes when completing the PR:
- bypassPolicy to true
- bypassReason with some valid reason

![image](https://github.com/renovatebot/renovate/assets/8171300/beda8024-490b-4c39-b9b1-5652f81010fe)
![image](https://github.com/renovatebot/renovate/assets/8171300/1dafa41d-0f76-4bec-a5b2-01db05417961)

Do you think this is something valuable to add in renovate bot?

Thanks for your help

Marc

**Comment by rarkins** [comment]
Won't this also bypass policies like "all tests must pass"?

**Reply by marcmognol** [reply]
You're right, using bypass policy option doesn't care about build status which in fact is not a good solution for us.
Is there any other way to achieve autoComplete in azure?

**Reply by rarkins** [reply]
Does https://docs.renovatebot.com/configuration-options/#autoapprove help?

**Reply by marcmognol** [reply]
No, autoApprove works perfect renovate approves but as my policy requires at least 2 reviewers, one reviewer is missing.

**Reply by viceice** [reply]
then you probably need a second account and approve with that automatically if it's a renovate pr. something similar like the auto approve bot for GitHub

**Reply by marcmognol** [reply]
Hi @viceice 

Yes, I can combine a webhook and an Azure function to bypass policy when:
- pr is a renovate branch (author + branch name)
- all build status passed

Thanks
